cocktails9/cocktails9/Views/Cocktails/CocktailsGridView.swift

```diff
@@ -0,0 +1,162 @@
+                                .padding(10)
+                                .background(Color(.systemGray6))
+                                .cornerRadius(10)
+                                .foregroundColor(.black)
+                        }
+                    }
+                    .padding(.horizontal)
+                    
+                    // Search Field (Dropdown)
+                    if isSearchVisible {
+                        TextField("Search cocktails", text: $searchText)
+                            .padding(10)
+                            .background(Color(.systemGray6))
+                            .cornerRadius(10)
+                            .overlay(
+                                HStack {
+                                    Spacer()
+                                    if !searchText.isEmpty {
+                                        Button(action: {
+                                            searchText = ""
+                                            Task {
+                                                await viewModel.performSearch()
+                                            }
+                                        }) {
+                                            Image(systemName: "xmark.circle.fill")
+                                                .foregroundColor(.gray)
+                                                .padding(.trailing, 8)
+                                        }
+                                    }
+                                }
+                            )
+                            .onChange(of: searchText) { newValue in
+                                Task {
+                                    viewModel.searchQuery = newValue
+                                    await viewModel.performSearch()
+                                }
+                            }
+                            .transition(.opacity.combined(with: .move(edge: .top)))
+                            .animation(.easeInOut, value: isSearchVisible)
+                            .padding(.horizontal)
+                    }
+                }
+                
+                // Cocktail Grid
+                if isLoading {
+                    ProgressView("Loading Cocktails...")
+                        .progressViewStyle(CircularProgressViewStyle())
+                        .padding(.top, 50)
+                } else if viewModel.cocktails.isEmpty {
+                    Text("No cocktails found.")
+                        .font(.title2)
+                        .foregroundColor(.gray)
+                        .padding(.top, 50)
+                } else {
+                    ScrollView {
+                        LazyVGrid(columns: columns, spacing: 15) {
+                            ForEach(viewModel.cocktails.filter { searchText.isEmpty || $0.name.localizedCaseInsensitiveContains(searchText) }, id: \.id) { cocktail in
```

Comment:
you don't need to filter searched local data here. Code from above (with `performSearch(:)`) will handle the UI